src/core/server/integration_tests/saved_objects/migrations/group2/multiple_kibana_nodes.test.ts

```diff
@@ -176,13 +176,21 @@ describe('migration v2', () => {
 
   const startWithDelay = async (instances: Root[], delayInSec: number) => {
     const promises: Array<Promise<unknown>> = [];
+    const errors: string[] = [];
     for (let i = 0; i < instances.length; i++) {
-      promises.push(instances[i].start());
+      promises.push(
+        instances[i].start().catch((err) => {
+          errors.push(err.message);
+        })
+      );
       if (i < instances.length - 1) {
         await delay(delayInSec * 1000);
       }
     }
-    return Promise.all(promises);
+    await Promise.all(promises);
+    if (errors.length) {
+      throw new Error(`Failed to start all instances: ${errors.join(',')}`);
+    }
```

Comment:
Promises pushed into the array could still throw while we were awaiting the `delay` calls to start the next instances, causing an unhandled rejection. We're now adding an error handler via `.catch` and then manually awaiting all and then throwing if any returned an error.